lib/js/components/DatePickers/DatePicker/DatePicker.composables.ts

```diff
@@ -130,6 +131,21 @@ export function initFlatpickr({
 			flush: 'post', // Ensure updates happen after DOM changes
 		},
 	);
+	watch(
+		[() => props.date],
```

Comment:
What about `startDate` and `endDate` in DateRangePicker? Is there a similar scenario? Maybe we should add it to the DatePickerComposablesProps interface as well